src/components/reusable/datePicker/datepicker.stories.js

```diff
@@ -1,114 +1,217 @@
+  // prevents flatpickr calendar overlay from persisting on view change
+  useEffect(() => {
+    return () => {
+      disconnectFlatpickr();
+    };
+  }, []);
+
+  const anchorId =
+    args.nameAttr ||
+    `date-picker-button-${Math.random().toString(36).slice(2, 11)}`;
+
+  const placeholder = getPlaceholder(args.dateFormat);
+
+  return html`<kyn-date-picker
+    .nameAttr="${args.nameAttr}"
+    .locale="${args.locale}"
+    .dateFormat="${args.dateFormat}"
+    ?required="${args.required}"
+    .value="${args.value}"
+    .warnText="${args.warnText}"
+    .invalidText="${args.invalidText}"
+    .altFormat=${args.altFormat}
+    .disable="${args.disable}"
+    .enable="${args.enable}"
+    .mode="${args.mode}"
+    .caption="${args.caption}"
+    ?datePickerDisabled="${args.datePickerDisabled}"
+    ?twentyFourHourFormat="${args.twentyFourHourFormat}"
+    .minDate="${args.minDate}"
+    .maxDate="${args.maxDate}"
+    @on-change=${(e) => action(e.type)(e)}
+  >
+    <label
+      slot="label"
+      class="label-text"
+      for=${anchorId}
+      ?disabled=${args.datePickerDisabled}
+    >
+      ${args.required
+        ? html`<abbr
+            class="required"
+            title=${args._textStrings?.requiredText || 'Required'}
+            aria-label=${args._textStrings?.requiredText || 'Required'}
+            >*</abbr
+          >`
+        : null}
+      ${args.unnamed}
+    </label>
+    <input
+      slot="anchor"
+      type="text"
+      id=${anchorId}
+      name=${args.nameAttr}
+      placeholder=${placeholder}
+      ?disabled=${args.datePickerDisabled}
+      ?required=${args.required}
+      aria-invalid=${args._isInvalid ? 'true' : 'false'}
+    />
+    <span slot="icon" style="line-height: 1;">${unsafeSVG(overflowIcon)}</span>
+  </kyn-date-picker>`;
+};
+
+const ButtonTemplate = (args) => {
+  useEffect(() => {
+    return () => {
+      disconnectFlatpickr();
+    };
+  }, []);
+
+  return html`<kyn-date-picker
+    .nameAttr="${args.nameAttr}"
+    .locale="${args.locale}"
+    .dateFormat="${args.dateFormat}"
+    ?required="${args.required}"
+    .value="${args.value}"
+    .warnText="${args.warnText}"
+    .invalidText="${args.invalidText}"
+    .altFormat=${args.altFormat}
+    .disable="${args.disable}"
+    .enable="${args.enable}"
+    .mode="${args.mode}"
+    .caption="${args.caption}"
+    ?datePickerDisabled="${args.datePickerDisabled}"
+    ?twentyFourHourFormat="${args.twentyFourHourFormat}"
+    .minDate="${args.minDate}"
+    .maxDate="${args.maxDate}"
+    @on-change=${(e) => action(e.type)(e)}
+  >
+    <label slot="label" class="label-text" ?disabled=${args.datePickerDisabled}>
+      ${args.required
+        ? html`<abbr
+            class="required"
+            title=${args._textStrings?.requiredText || 'Required'}
+            aria-label=${args._textStrings?.requiredText || 'Required'}
+            >*</abbr
+          >`
+        : null}
+      ${args.unnamed}
+    </label>
+    <kd-button
```

Comment:
A button anchor concept is interesting, but I don't understand how it can function correctly with a form if there is no `<input>`, and how the user is supposed to know the selected date without opening the picker.

Reply:
button options removed in datepicker, timepicker, daterangepicker, starting with an internal, single input element that fills 100% of parent/containing element

i can implement sizing (sm, md, lg) if we want to do that in this case.